### ISSUE TITLE
java.lang.module.FindException:  Unable to derive module descriptor for class-model-2.6.1.jar — Invalid module name: 'class' is not a Java identifier

### ISSUE THREAD

**Issue by lingocoder** [eclipse-ee4j/glassfish-hk2#477]
### Context 

I need to build, deploy and run from the command line, Spring Boot Jersey applications that are distributed as Java Platform Module System (JPMS) modules.

### The Problem

Running a Spring Boot Jersey application, with a command like this:

```
$ java --module-path ..:/home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar:/home/lingocoder/libs/org.springframework.boot/spring-boot-starter-jersey/2.1.8.RELEASE/spring-boot-starter-jersey-2.1.8.RELEASE.jar:... -m my.module/my.springboot.application.Main
```

...fails with this *`java.lang.module.FindException`*:

```
...
Error occurred during initialization of boot layer
java.lang.module.FindException:  Unable to derive module descriptor for: /home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar

Caused by: java.lang.IllegalArgumentException: class.model: Invalid module name: 'class' is not a Java identifier
```
### Steps to Reproduce

1. The same error occurs when running this *`jar`* command:

```
$ jar -f /home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar --describe-module
```
2. This is the result of passing that same *`class-model`* jar to *`jdeps`*:

```
$ jdeps --generate-module-info . /home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar
Exception in thread "main" java.lang.module.FindException: Unable to derive module descriptor for /home/lingocoder/libs/org.glassfish.hk2/class-model/2.6.1/class-model-2.6.1.jar
        at java.base/jdk.internal.module.ModulePath.readJar(ModulePath.java:646)
        at java.base/jdk.internal.module.ModulePath.readModule(ModulePath.java:329)
        at java.base/jdk.internal.module.ModulePath.scan(ModulePath.java:235)
        at java.base/jdk.internal.module.ModulePath.scanNextEntry(ModulePath.java:188)
        at java.base/jdk.internal.module.ModulePath.findAll(ModulePath.java:164)
        at jdk.jdeps/com.sun.tools.jdeps.ModuleInfoBuilder.<init>(ModuleInfoBuilder.java:85)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask$GenModuleInfo.run(JdepsTask.java:943)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:574)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:533)
        at jdk.jdeps/com.sun.tools.jdeps.Main.main(Main.java:49)
Caused by: java.lang.IllegalArgumentException: class.model: Invalid module name: 'class' is not a Java identifier
        at java.base/jdk.internal.module.Checks.requireModuleName(Checks.java:52)
        at java.base/java.lang.module.ModuleDescriptor$Builder.<init>(ModuleDescriptor.java:1535)
        at java.base/java.lang.module.ModuleDescriptor.newAutomaticModule(ModuleDescriptor.java:2417)
        at java.base/jdk.internal.module.ModulePath.deriveModuleDescriptor(ModulePath.java:506)
        at java.base/jdk.internal.module.ModulePath.readJar(ModulePath.java:642)
        ... 9 more
```

### Related Issues

* [***jna-platform,xalan, batik-script InvalidModuleDescriptorException***](http://bit.ly/jigsawdev1059479)
* [***java.lang.module.FindException under Java 9***](http://bit.ly/RESTEASY1752)
* [***Modular java (eg JDK11) doesn't work correctly with RESTEasy client***](http://bit.ly/RESTEASY1887)
* [***Automatic Module Name***](https://github.com/eclipse-ee4j/glassfish-hk2/issues/458)
* [***Invalid module-info.class in asm-repackaged***](https://github.com/eclipse-ee4j/glassfish-hk2/issues/410)
* [***ArrayIndexOutOfBoundsException at io.github.classgraph.Classfile.getConstantPoolStringOffset***](https://github.com/classgraph/classgraph/issues/319)
* [***java.lang.module.FindException: Automatic-Module-Name: org.xmlunit.xmlunit-core: Invalid module name: 'xmlunit-core' is not a Java identifier***](https://github.com/xmlunit/xmlunit/issues/136)
* [***Upgrade to Tomcat 9.0.26***](https://github.com/spring-projects/spring-boot/issues/18388)

### Environment

```
$ java -version
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment (build 12+33)
OpenJDK 64-Bit Server VM (build 12+33, mixed mode, sharing)
...
OS: Windows amd64
```

Please fix this? TIA.

**Comment by rkraneis** [comment]
Does this relate to #458?

**Comment by buko** [comment]
This is a severe issue. It prevents hk2 from being used easily in any modular application. The only way to fix this seems to be prevent class-model.jar from ending up on the module-path and putting it on the class-path and then updating all launch scripts. It would be fixed by #458 . Any progress on modularizing the hk2 jars correctly?

**Comment by lingocoder** [comment]
Yes it does.


-------- Original Message --------
Subject: Re:_[eclipse-ee4j/glassfish-hk2]_java.lang.module.Fin
dException:__Unable_to_derive_module_descriptor_for_class-model-2
.6.1.jar_—_Invalid_module_name:_'class'_is_not_a_Java
_identifier_(#477)
